### PR TITLE
feat: use link or url as image provider

### PR DESCRIPTION
**Pull request by Cenikl**
* Fixed login and register to return token
* Updated documentation
* Changed creating/updating image on a user or idea to url instead of base64

### CODE REVIEW

File: doc/api.yaml [hunk 1]
```diff
@@ -492,15 +479,92 @@ paths:
           $ref: '#/components/responses/404'
         '500':
           $ref: '#/components/responses/500'
-
-
+  /ideas/{idea_id}/picture:
+    post:
+      tags:
+        - Ideas
```
Comment (haritianaadriano) [inline doc/api.yaml:483]: Check we have two duplicated Idea tags

File: src/main/java/com/starteo/demo/endpoint/controller/FundController.java [hunk 2]
```diff
@@ -29,13 +29,21 @@ public class FundController {
...
+  public List<Fund> saveFundByIdea(
+      @PathVariable(name = "idea_id") String ideaString,
+      HttpServletRequest request,
+      @RequestBody CreateFund toCreate) {
     User userConnected = authService.whoami(request);
-    com.starteo.demo.repository.model.Fund created = fundService.saveFundByIdeaId(toCreate, ideaString, userConnected);
+    com.starteo.demo.repository.model.Fund created =
+        fundService.saveFundByIdeaId(toCreate, ideaString, userConnected);
     return List.of(fundMapper.toRest(created));
   }
```
Comment (haritianaadriano) [inline src/main/java/com/starteo/demo/endpoint/controller/FundController.java]: When you provide the id in path it's for updating a specific funds. To create an idenpotant endpoint it's better to use directly the /funds endpoint and it's in the payload that we will add the id or not to create or update funds.

File: src/main/java/com/starteo/demo/endpoint/controller/FundController.java [hunk 3]
```diff
@@ -29,13 +29,21 @@ public class FundController {
...
   @GetMapping("/funds/{idea_id}")
-  public List<Fund> getAllFundSOnIdea(@PathVariable("idea_id")String ideaString,HttpServletRequest request){
-    return fundService.getFundsByIdea(ideaService.getById(ideaString)).stream().map(fundMapper::toRest).toList();
+  public List<Fund> getAllFundSOnIdea(
+      @PathVariable("idea_id") String ideaString, HttpServletRequest request) {
+    return fundService.getFundsByIdea(ideaService.getById(ideaString)).stream()
+        .map(fundMapper::toRest)
+        .toList();
   }
 }
```
Comment (haritianaadriano) [inline src/main/java/com/starteo/demo/endpoint/controller/FundController.java]: i think it's better to begin from ideas resource as /ideas/{idea_id}/funds

File: src/main/java/com/starteo/demo/endpoint/controller/IdeaController.java [hunk 4]
```diff
@@ -9,58 +9,77 @@
...
+    return ideaService
+        .getIdeasByWeek(
+            InstanceTime.getCurrentMondayOfTheWeek(),
+            InstanceTime.getCurrentSundayOfTheWeek(),
+            page,
+            pageSize)
+        .stream()
+        .map(ideaMapper::toRest)
+        .toList();
+  }
```
Comment (haritianaadriano) [inline src/main/java/com/starteo/demo/endpoint/controller/IdeaController.java:58]: we can directly grouped it with the /ideas resources i'll help you with that

**Review by haritianaadriano — CHANGES_REQUESTED** [review]
Hello, thank you very much for your contribution 😁. I hope our working hard will be successfull

File: doc/api.yaml (hunk 1 above)
Comment (Cenikl) [reply doc/api.yaml:483]: Tags changed to "Funds" on resources involving funds and "idea" on ideas

Comment (haritianaadriano) [reply doc/api.yaml:483]: you can also resolve discussion if it's resolved

**Pull request merged**